tools/system/dmf-man/main.c

```diff
@@ -0,0 +1,511 @@
+/**
+ * @file main.c
+ * @brief dmf-man - DMOD Documentation Viewer
+ * 
+ * This tool displays documentation for DMOD modules in Markdown format
+ * with basic VT100 terminal formatting support.
+ */
+
+#define _POSIX_C_SOURCE 200809L
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <dirent.h>
+#include <ctype.h>
+#include "dmod.h"
+
+// Default paths
+#define DEFAULT_DMF_DIR "./dmf"
+
+// Environment variables
+#define ENV_DOC_DIR "DMOD_DOC_DIR"
+#define ENV_DMF_DIR "DMOD_DMF_DIR"
+
+// VT100 escape codes
+#define VT100_RESET "\033[0m"
+#define VT100_BOLD "\033[1m"
+#define VT100_DIM "\033[2m"
+#define VT100_ITALIC "\033[3m"
+#define VT100_UNDERLINE "\033[4m"
+#define VT100_REVERSE "\033[7m"
+
+// Colors
+#define VT100_BLACK "\033[30m"
+#define VT100_RED "\033[31m"
+#define VT100_GREEN "\033[32m"
+#define VT100_YELLOW "\033[33m"
+#define VT100_BLUE "\033[34m"
+#define VT100_MAGENTA "\033[35m"
+#define VT100_CYAN "\033[36m"
+#define VT100_WHITE "\033[37m"
+
+/**
+ * @brief Print usage message
+ */
+static void PrintUsage(const char* app_name) {
+    printf("Usage: %s [OPTIONS] <module_name>\n", app_name);
```

Comment:
wszędzie tam, gdzie możemy unikamy używania stdlib - zamiast tego staramy się  używać `SAL` api z dmod, czyli dla printf to będzie `Dmod_Printf`